src/main/java/com/sonymobile/jenkins/plugins/kerberossso/PluginImpl.java

```diff
@@ -127,6 +135,15 @@ private void removeFilter() throws ServletException {
     }
 
     /**
+     * Create and attach the filter.
+     * @throws ServletException Unable to add filter.
+     */
+    private void registerFilter() throws ServletException {
+        this.filter = new KerberosSSOFilter(createConfigMap(), new SpnegoKerberosAuthenticationFactory());
+        PluginServletFilter.addFilter(filter);
```

Comment:
Yes, the refactoring is fine, I just wanted to point out where my concern was about that a restart might still be required. But I trust your judgement :)